contracts/FragToken.sol

```diff
@@ -64,20 +68,36 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable {
 
         // add to current locked amount
         _locksAmount[msg.sender] = _locksAmount[msg.sender] + amount;
-        _locksBlock[msg.sender] = block.number;
+
+        if(period == Period.TwoWeeks)
+            _lockTime[msg.sender] = block.timestamp + (2 * _TIMELOCK);
+        
+        else if(period == Period.OneMonth)
+            _lockTime[msg.sender] = block.timestamp + (4 * _TIMELOCK);
+
+        else if(period == Period.ThreeMonths)
+            _lockTime[msg.sender] = block.timestamp + (12 * _TIMELOCK);
+
+        else if(period == Period.SixMonths)
+            _lockTime[msg.sender] = block.timestamp + (24 * _TIMELOCK);
+
+        else {
+            if(period == Period.OneYear)
```

Comment:
move this up and `else` should simply fail?

Reply:
I introduced an additional `require` instead to check the period in input. It is where it fails. I think it's cleaner and also more coherent with how we already handle the inputs.

Reply:
there is no clean code with ethereum contract, everything costs gas. see my above comment 😭😄

Reply:
I didn't consider gas optimization. My very bad. Let me try few different approaches. Thank you for your comment!